controllers/reconcile.go

```diff
@@ -595,6 +613,50 @@ func (r *MicroK8sControlPlaneReconciler) scaleDownControlPlane(ctx context.Conte
 	return ctrl.Result{Requeue: true}, nil
 }
 
+func getClusterAgentClient(machines []clusterv1.Machine, delMachine clusterv1.Machine) (*clusteragent.Client, error) {
+	opts := clusteragent.Options{
+		IgnoreNodeIPs: sets.NewString(),
+		// NOTE(hue): Port is hard coded according to
+		// https://github.com/canonical/cluster-api-control-plane-provider-microk8s/blob/v0.6.10/control-plane-components.yaml#L96-L102
+		Port: "30000",
+	}
+	// NOTE(hue): We want to pick a random machine's IP to call POST /dqlite/remove on its cluster agent endpoint.
+	// This machine should preferably not be the <delMachine> itself but this is not forced by Microk8s.
+	for _, addr := range delMachine.Status.Addresses {
+		opts.IgnoreNodeIPs.Insert(addr.Address)
+	}
+
+	clusterAgentClient, err := clusteragent.NewClient(machines, opts)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize cluster agent client: %w", err)
+	}
+
+	return clusterAgentClient, nil
+}
+
+// removeMicrok8sNode removes the node from
+func (r *MicroK8sControlPlaneReconciler) removeNodeFromDqlite(ctx context.Context, clusterAgentClient *clusteragent.Client, delMachine clusterv1.Machine) error {
+	var removeEp string
+	for _, addr := range delMachine.Status.Addresses {
+		if ip := net.ParseIP(addr.Address); ip != nil {
+			// NOTE(hue): Port is hard coded according to
+			// https://github.com/canonical/cluster-api-control-plane-provider-microk8s/blob/v0.6.10/control-plane-components.yaml#L96-L102
+			removeEp = fmt.Sprintf("%s:2379", addr.Address)
```

Comment:
What if they did not set portCompatibilityRemap to true? It may be possible for you to access the Microk8sControlPlane configuration from here and check.

Reply:
Thanks for catching this. What a nice comment.